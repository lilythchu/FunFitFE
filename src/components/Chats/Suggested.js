import React, {useState, useEffect} from 'react';
import { StyleSheet, Text, View, FlatList, Button } from 'react-native';
import { StoryItem } from './Stories';
import GenderAvatars from '../Profile/GenderAvatars';
import { getRecFrdURL } from '../../../api/client';
import axios from 'axios';

const Suggested = ({navigation, token}) => {
  const [friendIds, setFriendIds] = useState([]);

  const getRecFriendIds = () => {
    axios
      .get(getRecFrdURL, {headers: {"Authorization": `Bearer ${token}`}})
      .then(res => {
        setFriendIds(res.data);
      })
      .catch(err => {
        console.log(err);
      })
  }

  useEffect(() => { getRecFriendIds() }, []);

  return (
    <View style={{paddingLeft: 20, paddingBottom: 15}}>
      <FlatList 
        data={friendIds}
        keyExtractor={item => item.id}
        horizontal
        renderItem={({item}) => (
          <GenderAvatars id = {item} navigation={navigation} token={token} />
        )}
        showsHorizontalScrollIndicator={false}
      />
    </View>
  )
}

export default Suggested

const styles = StyleSheet.create({})