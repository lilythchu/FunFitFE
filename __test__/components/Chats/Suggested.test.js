import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Suggested from '../../../src/components/Chats/Suggested';

describe('Suggested friends', () => {
  test('renders correctly', () => {
    render(<Suggested />);
    expect(screen).toMatchSnapshot();
  })
})