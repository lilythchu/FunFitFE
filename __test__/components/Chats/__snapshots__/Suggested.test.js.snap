// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggested friends renders correctly 1`] = `
<View
  style={
    Object {
      "paddingLeft": 20,
      "paddingVertical": 5,
    }
  }
>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
